File: apps/api/plane/app/serializers/companies/company.py
from rest_framework import serializers
from plane.app.serializers.base import BaseSerializer
from plane.db.models import Company, CompanySettings, CompanyMemberRole


class CompanySettingsSerializer(BaseSerializer):
    class Meta:
        model  = CompanySettings
        fields = [
            "id", "annual_leave_days", "sick_leave_days",
            "carry_over_max_days", "carry_over_expiry_months",
            "leave_year_start", "weekend_policy", "public_holiday_region",
            "pay_cycle", "pay_day", "probation_period_days",
            "default_notice_period_days", "created_at", "updated_at",
        ]
        read_only_fields = ["id", "created_at", "updated_at"]


class CompanyLiteSerializer(BaseSerializer):
    """Minimal — for nesting inside People, Contracts, etc."""
    class Meta:
        model  = Company
        fields = ["id", "name", "trading_name", "logo", "status", "default_currency"]
        read_only_fields = ["id"]


class CompanySerializer(BaseSerializer):
    settings = CompanySettingsSerializer(read_only=True)

    class Meta:
        model  = Company
        fields = [
            "id", "name", "trading_name", "registration_number",
            "tax_id", "vat_number", "company_type", "logo",
            "country", "city", "registered_address", "billing_address",
            "status", "default_currency", "default_timezone",
            "settings", "created_at", "updated_at",
        ]
        read_only_fields = ["id", "created_at", "updated_at"]


class CompanyMemberRoleSerializer(BaseSerializer):
    member_detail = serializers.SerializerMethodField()

    class Meta:
        model  = CompanyMemberRole
        fields = ["id", "company", "member", "role", "member_detail", "created_at"]
        read_only_fields = ["id", "created_at"]

    def get_member_detail(self, obj):
        user = obj.member.member  # WorkspaceMember.member → User
        return {
            "id":           str(obj.member.id),
            "display_name": getattr(user, "display_name", user.email),
            "email":        user.email,
            "avatar":       getattr(user, "avatar", ""),
        }

File: packages/types/src/company.ts
export type TCompanyStatus = "active" | "inactive" | "dissolved";
export type TPayCycle = "monthly" | "bi_monthly" | "weekly";
export type TLeaveYearStart = "calendar" | "anniversary";
export type TWeekendPolicy = "exclude" | "include";
export type TCompanyMemberRole = "hr_manager" | "finance_manager" | "director";

export interface ICompanySettings {
  id: string;
  annual_leave_days: number;
  sick_leave_days: number;
  carry_over_max_days: number;
  carry_over_expiry_months: number;
  leave_year_start: TLeaveYearStart;
  weekend_policy: TWeekendPolicy;
  public_holiday_region: string;
  pay_cycle: TPayCycle;
  pay_day: number;
  probation_period_days: number;
  default_notice_period_days: number;
  created_at: string;
  updated_at: string;
}

export interface ICompany {
  id: string;
  name: string;
  trading_name: string;
  registration_number: string;
  tax_id: string;
  vat_number: string;
  company_type: string;
  logo: string;
  country: string;
  city: string;
  registered_address: string;
  billing_address: string;
  status: TCompanyStatus;
  default_currency: string;
  default_timezone: string;
  settings?: ICompanySettings;
  created_at: string;
  updated_at: string;
}

export interface ICompanyMemberRole {
  id: string;
  company: string;
  member: string;
  role: TCompanyMemberRole;
  member_detail: {
    id: string;
    display_name: string;
    email: string;
    avatar: string;
  };
  created_at: string;
}
